### PR TITLE
feat: Add `timeFormatType` field

### PR DESCRIPTION
**Pull request by lsh**
Allows for the use of `timeFormatType` similar to `numberFormatType`.

<details>
  <summary>📦 Published PR as canary version: <code>5.4.1--canary.8320.193f79b.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install vega-lite@5.4.1--canary.8320.193f79b.0
  # or 
  yarn add vega-lite@5.4.1--canary.8320.193f79b.0
  ```
</details>

### CODE REVIEW

File: src/compile/format.ts [hunk 1]
```diff
@@ -211,6 +206,9 @@ export function guideFormat(
 
   if (isFieldOrDatumDefForTimeFormat(fieldOrDatumDef)) {
     const timeUnit = isFieldDef(fieldOrDatumDef) ? normalizeTimeUnit(fieldOrDatumDef.timeUnit)?.unit : undefined;
+    if (timeUnit === undefined && config.customFormatTypes && config.timeFormatType) {
```
Comment (lsh) [inline src/compile/format.ts:209]: This section is problematic. Without it, the test with the label: _returns undefined for a field if custom timeFormatType is in the config_ fails. However, with it, the test labelled _returns prefers timeUnit to timeFormatType_ fails.

File: src/config.ts [hunk 2]
```diff
@@ -189,6 +189,15 @@ export interface VLOnlyConfig<ES extends ExprRef | SignalRef> {
    */
   timeFormat?: string;
 
+  /**
+   * [Custom format type](https://vega.github.io/vega-lite/docs/config.html#custom-format-type)
+   * for `config.numberFormat`.
+   *
+   * __Default value:__ `undefined` -- This is equilvalent to call D3-format, which is exposed as [`format` in Vega-Expression](https://vega.github.io/vega/docs/expressions/#format).
```
Comment (kanitw) [inline src/config.ts:196]: I know you haven't finished this PR but don't forget to update these docs. :)

File: src/compile/format.ts [hunk 3]
```diff
@@ -59,47 +59,46 @@ export function formatSignalRef({
...
+        return formatCustomType({
+          fieldOrDatumDef,
+          format: config.numberFormat,
+          formatType: config.numberFormatType,
+          expr,
+          config
+        });
+      }
+    }
+    if (type === 'temporal' && config.timeFormatType) {
```
Comment (kanitw) [inline src/compile/format.ts:82]: Don't we have to check for time unit here?

File: src/config.ts (hunk 2 above)
Comment (kanitw) [reply src/config.ts:196]: > This is equilvalent to call D3-format

D3-time-format? 

> which is exposed as [`format` in Vega-Expression](https://vega.github.io/vega/docs/expressions/#format)

`timeFormat`

File: src/compile/format.ts [hunk 4]
```diff
@@ -179,22 +178,16 @@ export function guideFormat(
...
-      fieldOrDatumDef.stack === 'normalize'
+      channelDefType(fieldOrDatumDef) === 'quantitative' &&
+      ((config.normalizedNumberFormatType &&
+        isPositionFieldOrDatumDef(fieldOrDatumDef) &&
+        fieldOrDatumDef.stack === 'normalize') || // case: normalized number format
+        config.numberFormatType) // case: regular number format
     ) {
       return undefined; // handled in encode block
     }
-    if (config.numberFormatType) {
```
Comment (kanitw) [inline src/compile/format.ts:195]: Why is this block removed?

Comment (kanitw) [reply src/compile/format.ts:195]: Oh I see, you collapse them into `||` but I think that makes it harder to read.  I'd rather keep the two blocks for readability.

**Review by kanitw — COMMENTED** [review]
Some minor comments, but the main one is don't we have to check for time unit in `formatSignalRef`?

**Review by kanitw — APPROVED** [review]

**Pull request merged**